docs/icicle/primitives/msm.md

```diff
@@ -1,19 +1,88 @@
-# MSM
+# MSM - Multi scalar multiplication
 
-MSM (Multi scalar multiplication) is an important primitive in ZK protocols.
+MSM stands for Multi scalar multiplication, its defined as:
 
-### Using MSM
+<math xmlns="http://www.w3.org/1998/Math/MathML">
+  <mi>M</mi>
+  <mi>S</mi>
+  <mi>M</mi>
+  <mo stretchy="false">(</mo>
+  <mi>a</mi>
+  <mo>,</mo>
+  <mi>G</mi>
+  <mo stretchy="false">)</mo>
+  <mo>=</mo>
+  <munderover>
+    <mo data-mjx-texclass="OP" movablelimits="false">&#x2211;</mo>
+    <mrow data-mjx-texclass="ORD">
+      <mi>j</mi>
+      <mo>=</mo>
+      <mn>0</mn>
+    </mrow>
+    <mrow data-mjx-texclass="ORD">
+      <mi>n</mi>
+      <mo>&#x2212;</mo>
+      <mn>1</mn>
+    </mrow>
+  </munderover>
+  <msub>
+    <mi>a</mi>
+    <mi>j</mi>
+  </msub>
+  <msub>
+    <mi>G</mi>
+    <mi>j</mi>
+  </msub>
+</math>
 
-#### Supported algorithms
+Where
 
-Our MSM implementation supports two algorithms.
+$G_j \in G$ - points from an Elliptic Curve group.
 
--   Bucket accumulation 
--   Large triangle accumulation
+$a_0, \ldots, a_n$ - Scalars
 
-In most scenarios you should be using Bucket accumulation as its a more parallel friendly algorithm and will yield better performance on larger MSMs. 
+$MSM(a, G) \in G$ - a single EC (elliptic curve point) point
 
-The exception being a large batch of smaller MSMs.
+Or in more simple terms MSM is the sum of scalar and EC point multiplications, we can learn from this definition as well that the core operations occurring are Modular Multiplication and Elliptic curve point addition. Since each multiplication can be computed independently and then the products need to be summed, this makes MSM inherently extremely parallelizable.
+
+Accelerating MSM is crucial to a ZK protocol performance due to the [large percent of run time](https://hackmd.io/@0xMonia/SkQ6-oRz3#Hardware-acceleration-in-action) they take up when generating proofs.
+
+You can learn more about how MSMs work from this [video](https://www.youtube.com/watch?v=Bl5mQA7UL2I) and from this resource list on [Ingopedia](https://www.ingonyama.com/ingopedia/msm).
+
+# Using MSM
+
+## Supported curves
+
+MSM supports all curves;
+
+`bls12-377`, `bls12-381`, `bn-254`, `bw6-761`
+
+## Supported algorithms
+
+Our MSM implementation supports two algorithms `Bucket accumulation` and `Large triangle accumulation`.
+
+### Bucket accumulation
+
+The Bucket Accumulation algorithm is a method of dividing the overall MSM task into smaller, more manageable sub-tasks. It involves partitioning scalars and their corresponding points into different "buckets" based on the scalar values.
+
+Bucket Accumulation can be more parallel-friendly because it involves dividing the computation into smaller, independent tasks, distributing scalar-point pairs into buckets and summing points within each bucket. This division makes it well-suited for parallel processing on GPUs.
```

Comment:
```suggestion
Bucket Accumulation can be more parallel-friendly because it involves dividing the computation into smaller, independent tasks, distributing scalar-point pairs into buckets and summing points within each bucket. This division makes it well suited for parallel processing on GPUs.
```